src/napari/_qt/layer_controls/widgets/_vectors/qt_line_dimension_spinbox.py

```diff
@@ -0,0 +1,133 @@
+        self.width_spinbox.setValue(self._layer.edge_width)
+        self.width_spinbox.valueChanged.connect(self.change_width)
+
+        self.width_spinbox_label = QtWrappedLabel(trans._('width:'))
+
+    def change_width(self, value) -> None:
+        """Change edge line width of vectors on the layer model.
+
+        Parameters
+        ----------
+        value : float
+            Line width of vectors.
+        """
+        self._layer.edge_width = value
+        self.width_spinbox.clearFocus()
+        # TODO: Check other way to give focus without calling parent
+        self.parent().setFocus()
+
+    def _on_edge_width_change(self) -> None:
+        """Receive layer model width change event and update width spinbox."""
+        with self._layer.events.edge_width.blocker():
+            self.width_spinbox.setValue(self._layer.edge_width)
+
+    def get_widget_controls(self) -> list[tuple[QtWrappedLabel, QWidget]]:
+        return [(self.width_spinbox_label, self.width_spinbox)]
+
+
+class QtLengthSpinBoxControl(QtWidgetControlsBase):
+    """
+    Class that wraps the connection of events/signals between the layer length of vectors
+    attribute and Qt widgets.
+
+    Parameters
+    ----------
+    parent: qtpy.QtWidgets.QWidget
+        An instance of QWidget that will be used as widgets parent
+    layer : napari.layers.Layer
+        An instance of a napari layer.
+
+    Attributes
+    ----------
+    length_spinbox : qtpy.QtWidgets.QDoubleSpinBox
+        Spinbox widget controlling line length of vectors.
+        Multiplicative factor on projections for length of all vectors.
+    length_spinbox_label : napari._qt.layer_controls.widgets.qt_widget_controls_base.QtWrappedLabel
+        Label for line length of vectors chooser widget.
+    """
+
+    def __init__(self, parent: QWidget, layer: Layer) -> None:
+        super().__init__(parent, layer)
+        # Setup layer
+        self._layer.events.length.connect(self._on_length_change)
+
+        # Setup widgets
+        # line length
+        self.length_spinbox = QDoubleSpinBox()
+        self.length_spinbox.setKeyboardTracking(False)
+        self.length_spinbox.setSingleStep(0.1)
+        self.length_spinbox.setValue(self._layer.length)
+        self.length_spinbox.setMinimum(0.1)
+        self.length_spinbox.setMaximum(np.inf)
+        self.length_spinbox.valueChanged.connect(self.change_length)
+
+        self.length_spinbox_label = QtWrappedLabel(trans._('length:'))
+
+    def change_length(self, value: float) -> None:
+        """Change length of vectors on the layer model.
+
+        Multiplicative factor on projections for length of all vectors.
+
+        Parameters
+        ----------
+        value : float
+            Length of vectors.
+        """
+        self._layer.length = value
+        self.length_spinbox.clearFocus()
+        # TODO: Check other way to give focus without calling parent
+        self.parent().setFocus()
```

Comment:
why does this have this extra clearfocus? Is it needed? If not, this can be `connect_setattr`

Reply:
Looks like this stuff was added way back in #536 to fix issues like #500. Maybe we shoulld just add this clearfocus stuff directly to `connect_setattr`...